packages/lib/src/components/Pix/PixInput/PixInput.tsx

```diff
@@ -0,0 +1,103 @@
+import {h} from 'preact';
+import {useEffect, useRef, useState} from 'preact/hooks';
+import {renderFormField} from '../../internal/FormFields';
+import Field from '../../internal/FormFields/Field';
+import {pixValidationRules} from './validate';
+import {pixFormatters} from './utils';
+import useCoreContext from '../../../core/Context/useCoreContext';
+import {PixInputDataState} from '../types';
+import useForm from '../../../utils/useForm';
+import SocialSecurityNumberBrazil from "../../internal/SocialSecurityNumberBrazil/SocialSecurityNumberBrazil";
+
+function PixInput(props) {
+    const {i18n} = useCoreContext();
+    const addressRef = useRef(null);
+    const formSchema = ['firstName', 'lastName', 'socialSecurityNumber'];
+    const {handleChangeFor, triggerValidation, setSchema, data, valid, errors, isValid} = useForm<PixInputDataState>({
+        schema: formSchema,
+        defaultData: props.data,
+        rules: pixValidationRules,
+        formatters: pixFormatters
+    });
+
+    // Handle form schema updates
+    useEffect(() => {
+        const newSchema = [
+            ...(props.personalDetailsRequired ? formSchema : []),
+        ];
+        setSchema(newSchema);
+    }, [props.personalDetailsRequired]);
+
+
+    const [status, setStatus] = useState('ready');
+    this.setStatus = setStatus;
+
+    this.showValidation = () => {
+        triggerValidation();
+        if (props.billingAddressRequired) {
```

Comment:
Is this correct? Should this be `personalDetailsRequired`?

Reply:
It wasn't needed